### ISSUE TITLE
Funnel on Client using "-tun userspace-networking"

### ISSUE THREAD

**Issue by dengesCU** [tailscale/tailscale#12788]
### What is the issue?

`tailscale funnel` returns output that suggests that everything works, i.e.,

> Available on the internet:
> 
> https://***.ts.net/
> |-- proxy http://127.0.0.1:8080
> 
> Funnel started and running in the background.
> To disable the proxy, run: tailscale funnel --https=443 off

but when opening the URL in web browser, it shows an SSL_ERROR_INTERNAL_ERROR_ALERT error.
When running ` tailscale cert <domain>`, I get the following error message:

> 500 Internal Server Error: acme.GetReg: Get "https://acme-v02.api.letsencrypt.org/directory": dial tcp: lookup acme-v02.api.letsencrypt.org on [::1]:53: read udp [::1]:32884->[::1]:53: read: connection refused

I believe the problem is that the way `tailscale cert` works is that it changes the DNS settings, which is not possible without root privileges. Thus, it should be expected that `tailscale funnel` does not work when using the user-space networking option as a non-root user. However, I could not find this documented anywhere. If I'm right about this, then I would find a warning message when running tailscale funnel while using userspace-networking useful.

### OS

Linux, Android

### OS version

Termux 0.118.1

### Tailscale version

1.68.2

### Bug report

BUG-0059d197d93c98153ee2cc8b73c4d221740b258c6c721fc8c4689c56f1f8ba69-20240712090106Z-3c201cad6dc70dd6

**Comment by bradfitz** [comment]
Sorry, this slipped through the cracks.

Getting a cert doesn't require changing your client's DNS settings; the DNS challenge for LetsEncrypt/ACME is done on our servers and doesn't depend on userspace mode or not on your client. In fact, we regularly use Funnel in userspace mode.

It looks like something is blocking your access to LetsEncrypt. Is there a firewall blocking that?

**Comment by dengesCU** [comment]
To be fair, this happened in a bit of an esoteric environment (Tailscale binary within a non-root Termux installation on Android).  Unfortunately, I cannot test on the device any longer since the battery started swelling a few months back..

That being said the error message to me read as port 53 on localhost refused the connection because no local DNS is running and Tailscale also cannot change any DNS settings. I will check whether I can do tailscale cert on the network on an other device ( I am pretty sure I can ), so I still think the error is due to running Tailscale as unprivileged user. 

But I agree with you, it's certainly not a general issue with userpace networking. In Docker containers funnel and serve works without any issue. Thank you for your answer!

**Comment by bradfitz** [comment]
> non-root Termux installation on Android

Hah, yeah... that's almost certainly it.

We don't do any effort or testing on that at all. And that environment is very funky.

**Comment by dengesCU** [comment]
Yeah I won't argue with that. I am already amazed by how well Tailscale works in there. Thanks again
